frontend/src/metabase/components/Schedule/components.tsx

```diff
@@ -164,26 +180,38 @@ export const SelectMinute = ({
   );
 };
 
+const estimateWidthOfString_base = (str: string) => {
+  const canvas = document.createElement("canvas");
+  const context = canvas.getContext("2d");
+  if (!context) {
+    return 0;
+  }
+  context.font = "14px sans-serif";
+  const width = context.measureText(str).width;
+  return width;
+};
+export const estimateWidthOfString = _.memoize(estimateWidthOfString_base);
+
 export const AutoWidthSelect = (props: SelectProps) => {
   const longestLabel = useMemo(
     () => getLongestSelectLabel(props.data),
     [props.data],
   );
-  const maxWidth =
-    longestLabel.length > 15 ? "unset" : `${longestLabel.length + 0.85}rem`;
+  const maxWidth = `${estimateWidthOfString(longestLabel) + 60}px`;
```

Comment:
Few things:
1. Why is this needed? I'm a bit concerned that this is adding complexity that manual setting a wider input could also solve.
2. I don't know if we want to use a canvas to measure? i think we could probably use a div for the same purpose
3. canvas creation can fail, we're not handling the error
4. your memoization function on line 193 is a memory leak if we begin to use this lots of places. I would suggest creating a memoized instance of the memoized function within the AutoSelect component (i.e. `useMemo(() => _.memoize(estimateWidthOfString_base))`) so that it gets gc'ed on unmount.

Reply:
1. In many cases it's nice to have the selects appear on the same line if possible, like this:

![image](https://github.com/user-attachments/assets/ced18ddc-b5fd-4358-8b65-e606043646f0)

2. This approach doesn't change the DOM so some people prefer it ([discussion on SO](https://stackoverflow.com/questions/118241/calculate-text-width-with-javascript/21015393#21015393))

3. Just in case there are any issues I've wrapped it in a try/catch with a fallback value. It has good support: [caniuse](https://caniuse.com/mdn-api_canvasrenderingcontext2d_measuretext)

4. This is a really interesting - and cool - point. I've pushed a change along these lines; let's discuss it